backend/layers/business/business.py

```diff
@@ -0,0 +1,190 @@
+    # Returns a handle to the revised collection (either id or the full collection metadata)
+
+    def create_collection_version(self, collection_id: str, user_info: UserInfo) -> CollectionVersion:
+        pass
+
+    def delete_collection_version(self, version_id: str, user_info: UserInfo) -> None:
+        pass
+
+
+    # Publish_collection
+    # Replaces post (in publish.py)
+    # Accepts:
+    # Collection_id
+    # UserInfo
+    # Performs validation to make sure that the collection can be published
+    # [Currently] accepts data_submission_policy_version: what is this for?
+    # [Currently] triggers Cloudfront invalidation for the index endpoints. This should arguably NOT be done here but by the API layer
+    # Since revision logic is database specific, it delegates to the underlying layer
+
+    def publish_collection_version(self, version_id: str, user_info: UserInfo) -> None:
+        pass
+
+    # Ingest_dataset
+    # Replaces the existing Upload_from_link
+    # Potentially, also replaces relink (I am not sure why they are 2 separate functions)
+    # Accepts:
+    # Collection_id
+    # UserInfo
+    # URL of the uploadable dataset
+    # [Optional] a dataset_id to be replaced
+    # This is one of the most complex functions. Other than the database provider, It will need two additional providers:
+    # StepFunctionProvider (to call the SFN that triggers the upload)
+    # DropboxProvider to interface with Dropbox (could be more generic: RemoteFileProvider?)
+    # Should handle exceptions from all providers:
+    # Should only raise custom exceptions
+
+    def ingest_dataset(self, collection_version_id: str, url: str, existing_dataset_version_id: Optional[str], user_info: UserInfo) -> str:
+        pass
+
+    # Get_all_datasets
+    # Replaces get_dataset_index
+
+    def get_all_datasets(self) -> Iterable[DatasetVersion]:
+        pass
+
+    # Delete_dataset
+    # Replaces delete_dataset
+
+    def delete_dataset(self, dataset_version_id: str) -> None:
+        pass
+
+
+    # get_dataset_assets
+    # Replaces get_dataset_assets
+
+    def get_dataset_artifacts(self, dataset_id: str) -> Iterable[DatasetArtifact]:
+        pass
+
+
+    # Download_dataset_asset
+    # Replaces post_dataset_asset
+
+
+    def download_dataset_asset(self, dataset_id: str) -> str:
+        pass
+
+    # Get_dataset_status
+    # Replaces get_status
+
+    def get_dataset_status(self, dataset_id: str) -> DatasetStatus:
+        pass
+
+
+# TODO: move it to a separate file
+class BusinessLogic(BusinessLogicInterface):
+
+    database_provider: DatabaseProviderInterface
+    crossref_provider: CrossrefProviderInterface
+    step_function_provider: StepFunctionProviderInterface
+
+    def __init__(
+        self, 
+        database_provider: DatabaseProviderInterface, 
+        crossref_provider: CrossrefProviderInterface,
+        step_function_provider: StepFunctionProviderInterface,
```

Comment:
is there anything to initialize in `BusinessLogicInterface`, else remove `super().__init__()`

Reply:
I will get rid of it, thanks. I think @atolopko-czi suggested to use [ABC](https://docs.python.org/3/library/abc.html) which would make interfaces look neater.